packages/aws-cdk/lib/logging.ts

```diff
@@ -7,6 +7,14 @@ const { stdout, stderr } = process;
 
 type WritableFactory = () => Writable;
 
+let LOG_LOCK = false;
+
+export function setLogLock(lock: boolean) {
+  LOG_LOCK = lock;
```

Comment:
Using a boolean is not re-entrancy safe. Make it a counter, increment by 1 every time someone locks and decrement by 1 every time someone unlocks.